xc/common/utils/prjxray_create_ioplace.py

```diff
@@ -96,46 +96,58 @@ def main():
     # to be used in fasm2bels.
     iostandard_constraints = {}
 
-    fname = args.pcf.name.replace(".pcf", ".json")
-    if os.path.isfile(fname):
-        with open(fname, "r") as fp:
-            iostandard_constraints = json.load(fp)
-
+    if args.pcf:
+        fname = args.pcf.name.replace(".pcf", ".json")
+        if os.path.isfile(fname):
+            with open(fname, "r") as fp:
+                iostandard_constraints = json.load(fp)
+    net_to_pad = io_place.net_to_pad
+    if args.pcf:
+        pcf_constraints = parse_simple_pcf(args.pcf)
+        net_to_pad |= [(constr.net, constr.pad) for constr in pcf_constraints]
```

Comment:
I don't believe this is legal python?  `net_to_pad` is a set, and the list on the right cannot be merged like that:
```
>>> a = set([1, 2, 3])
>>> a |= [i for i in range(3, 6)]
Traceback (most recent call last):
  File "<console>", line 1, in <module>
TypeError: unsupported operand type(s) for |=: 'set' and 'list'
```

Changing the `[...]` to set(...)` would work.

Reply:
Fixed